src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php

```diff
@@ -148,29 +149,33 @@ protected function getEffectivePlugins(array $resources): array
      */
     protected function extractEffectivePlugins($effectivePlugins, $pluginsPerExporter): array
     {
-        foreach ($effectivePlugins as $effectivePlugin) {
+        foreach ($effectivePlugins as $effectiveEventPlugins) {
+            $effectivePlugin = $this->resolveDuplicatePlugins($effectiveEventPlugins);
             if ($effectivePlugin instanceof EventResourceRepositoryPluginInterface || $effectivePlugin instanceof EventResourceBulkRepositoryPluginInterface) {
-                $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
+                $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS][] = $effectivePlugin;
             }
 
             if ($effectivePlugin instanceof EventResourceQueryContainerPluginInterface) {
-                $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
+                $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS][] = $effectivePlugin;
             }
         }
 
-        $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS] = $this->resolveDuplicatePlugins($pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS]);
-        $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS] = $this->resolveDuplicatePlugins($pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS]);
-
         return $pluginsPerExporter;
     }
 
     /**
      * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[] $eventResourcePlugins
      *
-     * @return \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[]
+     * @return \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface
      */
-    protected function resolveDuplicatePlugins(array $eventResourcePlugins): array
+    protected function resolveDuplicatePlugins(array $eventResourcePlugins): EventResourcePluginInterface
     {
-        return array_map('current', $eventResourcePlugins);
+        foreach ($eventResourcePlugins as $eventResourcePlugin) {
+            if ($eventResourcePlugin instanceof EventResourceRepositoryPluginInterface || $eventResourcePlugin instanceof EventResourceBulkRepositoryPluginInterface) {
```

Comment:
It will be better to move this condition to the protected method.
And interfaces to the `const`.

Reply:
Do you mean the interfaces like `EventResourceRepositoryPluginInterface` they are constants already, not strings :)

I didn't get it, what should I move to a protected method exactly?

Reply:
~~~php
protected function a(EventResourcePluginInterface $eventResourcePlugin): bool
{
     return $eventResourcePlugin instanceof EventResourceRepositoryPluginInterface
          || $eventResourcePlugin instanceof EventResourceBulkRepositoryPluginInterface
}
~~~
or even

~~~php
protected function a(EventResourcePluginInterface $eventResourcePlugin): bool
{
     foreach(static::CONST_WITH_PLUGINS as $pluginInterface) {
          if($eventResourcePlugin instanceof $pluginInterface) {
               return $eventResourcePlugin;
          }

          return null;
     }
}
~~~

Reply:
the second one example is for `open-closed principle`

Reply:
ok

Reply:
Done.